src/webui/components/Header/index.js

```diff
@@ -1,69 +1,179 @@
+    this.handleCloseInfoDialog = this.handleCloseInfoDialog.bind(this);
+    this.handleToggleLogin = this.handleToggleLogin.bind(this);
+    this.renderInfoDialog = this.renderInfoDialog.bind(this);
+    this.state = {
+      anchorEl: null,
+      openInfoDialog: false,
+      registryUrl: '',
+    };
+  }
+
+  componentDidMount() {
+    const registryUrl = getRegistryURL();
+    this.setState({
+      registryUrl,
+    });
+  }
+
+  handleMenu(event: MouseEvent<HTMLElement>) {
+    this.setState({
+      anchorEl: event.currentTarget,
+    });
+  }
+
+  handleClose() {
+    this.setState({
+      anchorEl: null,
+    });
+  }
+
+  handleOpenInfoDialog() {
+    this.setState({
+      openInfoDialog: true,
+    });
+  }
+
+  handleCloseInfoDialog() {
+    this.setState({
+      openInfoDialog: false,
+    });
+  }
+
+  handleToggleLogin() {
+    const {toggleLoginModal} = this.props;
+    this.setState(
+      {
+        anchorEl: null,
+      },
+      () => toggleLoginModal()
+    );
+  }
+
+  renderLeftSide() {
+    return (
+      <Link to="/">
+        <Logo />
+      </Link>
+    );
+  }
+
+  renderRightSide() {
+    const {username = ''} = this.props;
+    return (
+      <div>
+        <IconButton color="inherit" component={Link} to="https://verdaccio.org/docs/en/installation" blank>
+          <Help />
+        </IconButton>
+        <IconButton color="inherit" onClick={this.handleOpenInfoDialog}>
+          <Info />
+        </IconButton>
+        {username ? (
+          this.renderMenu()
+        ) : (
+          <Button color="inherit" onClick={this.handleToggleLogin}>
+            Login
+          </Button>
+        )}
+      </div>
+    );
+  }
+
+  renderMenu() {
+    const {username = '', handleLogout} = this.props;
+    const {anchorEl} = this.state;
+    const open = Boolean(anchorEl);
+    return (
+      <React.Fragment>
+        <IconButton aria-owns={username ? 'sidebar-menu' : null} aria-haspopup="true" color="inherit" onClick={this.handleMenu}>
+          <AccountCircle />
+        </IconButton>
+        <Menu
+          id="sidebar-menu"
+          anchorEl={anchorEl}
+          anchorOrigin={{
+            vertical: 'top',
+            horizontal: 'right',
+          }}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'right',
+          }}
+          open={open}
+          onClose={this.handleClose}
+        >
+          <MenuItem onClick={handleLogout}>Logout</MenuItem>
+        </Menu>
+      </React.Fragment>
+    );
+  }
+
+  renderCopyToClibBoard(text: string) {
+    return (
+      <ClipBoardCopy>
+        <ClipBoardCopyText>{text}</ClipBoardCopyText>
+        <Tooltip title="Copy to Clipboard">
+          <CopyIcon aria-label="Copy to Clipboard" onClick={copyToClipBoard(text)}>
+            <FileCopy />
+          </CopyIcon>
+        </Tooltip>
+      </ClipBoardCopy>
+    );
+  }
+
+  renderInfoDialog() {

```

Comment:
may be `rednerRegistryInformationDialog` ?

Reply:
It will host more than that, `renderInfoDialog` is ok in this case.